Research/Research/PlatformContext/RSDImageRules.swift

```diff
@@ -0,0 +1,277 @@
+        #else
+        let imageName = resourceName
+        #endif
+        
+        return rBundle.path(forResource: imageName, ofType: ext)
+    }
+    
+    
+    /// As of this writing, there is a bug in some versions of of iOS supported by this framework
+    /// that causes larger images to crash the application b/c of low memory and improper release
+    /// of the cache. As a result of this issue, large images should *not* be stored in the asset
+    /// catalog. Instead, they should be stored as files using the iOS convention of `imageName@2x`
+    /// and `imageName@2x` so that they can be created using the `UIImage(contentsOfFile:)` method.
+    /// syoung 12/12/2019
+    ///
+    /// This method is written to look *first* for the image using the `UIImage(contentsOfFile:)`
+    /// initializer. If not found, it will then look for the image in the asset bundles.
+    ///
+    /// - seealso: https://forums.developer.apple.com/thread/18767
+    ///
+    /// - parameters:
+    ///     - resourceName: The name of the image.
+    ///     - resourceInfo: The resource info associated with this asset.
+    ///     - traitCollection: The trait collection for the presenting view.
+    /// - returns: Image if found.
+    open func largeImage(named resourceName: String,
+                         using resourceInfo: RSDResourceInfo? = nil,
+                         compatibleWith traitCollection: UITraitCollection? = nil) -> RSDImage? {
+        
+        return _getFileImage(resourceName, resourceInfo, traitCollection) ??
+            _getAssetImage(resourceName, resourceInfo, traitCollection)
+    }
+    
+    /// Returns the image asset (if found).
+    ///
+    /// If `shouldPreferSystemImage == true` this method will look for an image in the SF images
+    /// first and *then* look for it in the resource info, main bundle, and registered bundles.
+    /// If `shouldPreferSystemImage == false`, then it will search the bundles and *then* look for
+    /// the asset in the SF image catalog.
+    ///
+    /// - parameters:
+    ///     - resourceName: The name of the image.
+    ///     - resourceInfo: The resource info associated with this asset.
+    ///     - traitCollection: The trait collection for the presenting view.
+    /// - returns: Image if found.
+    open func assetImage(named resourceName: String,
+                    using resourceInfo: RSDResourceInfo? = nil,
+                    compatibleWith traitCollection: UITraitCollection? = nil) -> RSDImage? {
+        
+        if shouldPreferSystemImage {
+            return _getSystemImage( resourceName, traitCollection) ??
+                _getAssetImage(resourceName, resourceInfo, traitCollection)
+        }
+        else {
+            return _getAssetImage(resourceName, resourceInfo, traitCollection) ??
+                _getSystemImage( resourceName, traitCollection)
+        }
+    }
+    
+    /// Private getter that can be used to check for availability.
```

Comment:
I'm thinking private method comments shouldn't be doc comments.
```suggestion
    // Private getter that can be used to check for availability.
```

Reply:
Doc comments can be automated to include/exclude private and/or internal methods. Since those often change level when needs of the developers change, I tend to start with private, but open later. That said, I'll change the comment to *not* reference private.